service: migration_manager: change the prepare_ methods to functions

patjed41 opened the pull request:
The `migration_manager` service is responsible for schema convergence in the cluster - pushing schema changes to other nodes and pulling schema when a version mismatch is observed. However, there is also a part of `migration_manager` that doesn't really belong there - creating mutations for schema updates. These are the functions with `prepare_` prefix. They don't modify any state and don't exchange any messages. They only need to read the local database.

We take these functions out of `migration_manager` and make them separate functions to reduce the dependency of other modules (especially `query_processor` and CQL statements) on `migration_manager`. Since all of these functions only need access to `storage_proxy` (or even only `replica::database`), doing such a refactor is not complicated. We just have to add one parameter, either `storage_proxy` or `database` and both of them are easily accessible in the places where these functions are called.

This refactor makes `migration_manager` unneeded in a few functions:
- `alternator::executor::create_keyspace`,
- `cql3::statements::alter_type_statement::prepare_announcement_mutations`,
- `cql3::statements::schema_altering_statement::prepare_schema_mutations`,
- `cql3::query_processor::execute_thrift_schema_command:`,
- `thrift::handler::execute_schema_command`.

We remove the `migration_manager&` parameter from all these functions.

Fixes #14339

scylladb-promoter commented:
CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2735/

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2741/

kbr-scylla reviewed (commented):
Nice cleanup, thanks. Left some small comments.

patjed41 commented on service/migration_manager.cc:
```diff
@@ -644,381 +644,118 @@ public void notifyDropAggregate(UDAggregate udf)
 }
 #endif
 
-std::vector<mutation> prepare_keyspace_update_announcement(replica::database& db, lw_shared_ptr<keyspace_metadata> ksm, api::timestamp_type ts) {
```
> If you give me a good reason for this movement (better than "their placements in .hh and .cc have been mismatched" -- I don't really understand why this would be a problem) then I won't complain.

I don't have a better reason. I've removed this commit.

> I don't think this commit is necessary, in fact it has a drawback: it will be harder to find the authors of logical changes in this code through git blame.

Right, I didn't think about it. I'll keep that in mind in the future.

patjed41 commented on cql3/statements/alter_type_statement.cc:
```diff
@@ -47,9 +47,9 @@ const sstring& alter_type_statement::keyspace() const
     return _name.get_keyspace();
```
Nice, I missed the `data_dictionary()` getter. I changed it as you suggested.

patjed41 commented:
v2:
- removed the commit that moved the `prepare_` functions to the end of `migration_manager.cc`
- `alter_type_statement::prepare_announcement_mutations`: changed a parameter from `query_processor&` to `storage_proxy&`

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2791/
